tests/probes_test.go

```diff
@@ -138,7 +146,7 @@ var _ = Describe("[ref_id:1182]Probes", func() {
 			tests.WaitForSuccessfulVMIStartIgnoreWarnings(vmi)
 
 			By("Starting the server inside the VMI")
-			serverStarter(vmi, 1500)
+			serverStarter(vmi, 1500, false)
```

Comment:
How come the livenessprobe succeed while the readiness fails. Doesn't it use the same nc http server?

Reply:
It does use the same `nc` server, and I don't fully understand what is happening, but, the behavior is consistent, check the following logs below:

Kubelet logs for readiness test:
```
Apr 20 16:51:13 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:13.227476    8244 prober.go:116] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" failed (failure): Get http://[fd00:10:244::27]:1500/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Apr 20 16:51:18 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:18.227726    8244 prober.go:116] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" failed (failure): Get http://[fd00:10:244::27]:1500/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Apr 20 16:51:22 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:22.234961    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:48896->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:51:27 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:27.235121    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:48954->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:51:32 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:32.236411    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:48990->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:51:37 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:37.234607    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49048->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:51:42 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:42.234709    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49084->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:51:47 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:47.234601    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49170->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:51:52 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:52.234071    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49206->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:51:57 kind-1.17.0-control-plane kubelet[8244]: I0420 16:51:57.234612    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49266->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:52:02 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:02.235071    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49302->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:52:07 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:07.234810    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49358->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:52:12 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:12.234748    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49394->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:52:17 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:17.234937    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49472->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:52:22 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:22.234477    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49532->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:52:27 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:27.235098    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49594->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:52:32 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:32.229708    8244 prober.go:111] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" errored: read tcp [fd00:10:244::1]:49630->[fd00:10:244::27]:1500: read: connection reset by peer
Apr 20 16:52:38 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:38.227644    8244 prober.go:116] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" failed (failure): Get http://[fd00:10:244::27]:1500/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Apr 20 16:52:43 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:43.227546    8244 prober.go:116] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" failed (failure): Get http://[fd00:10:244::27]:1500/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Apr 20 16:52:48 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:48.227908    8244 prober.go:116] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" failed (failure): Get http://[fd00:10:244::27]:1500/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Apr 20 16:52:53 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:53.227720    8244 prober.go:116] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" failed (failure): Get http://[fd00:10:244::27]:1500/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Apr 20 16:52:58 kind-1.17.0-control-plane kubelet[8244]: I0420 16:52:58.227695    8244 prober.go:116] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" failed (failure): Get http://[fd00:10:244::27]:1500/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
Apr 20 16:53:03 kind-1.17.0-control-plane kubelet[8244]: I0420 16:53:03.227449    8244 prober.go:116] Readiness probe for "virt-launcher-testvmi7862tkcmsj6495dzngjlnckgrggz4nhtj5m7bnbgv4_kubevirt-test-default(b0204a81-3319-497e-b2a4-12c939dd26e1):compute" failed (failure): Get http://[fd00:10:244::27]:1500/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

Now kubelet's logs for liveness check: 
```
Apr 20 16:59:57 kind-1.17.0-control-plane kubelet[8244]: I0420 16:59:57.079546    8244 prober.go:111] Liveness probe for "virt-launcher-testvmib2nfdbzw67zkn6svftrlh2tg7zrdcgr2wgw6lwz86x_kubevirt-test-default(c64c9157-c7a1-43ff-ba10-64468e7b3415):compute" errored: read tcp [fd00:10:244::1]:59054->[fd00:10:244::28]:1500: read: connection reset by peer
Apr 20 17:00:02 kind-1.17.0-control-plane kubelet[8244]: I0420 17:00:02.078982    8244 prober.go:111] Liveness probe for "virt-launcher-testvmib2nfdbzw67zkn6svftrlh2tg7zrdcgr2wgw6lwz86x_kubevirt-test-default(c64c9157-c7a1-43ff-ba10-64468e7b3415):compute" errored: read tcp [fd00:10:244::1]:59090->[fd00:10:244::28]:1500: read: connection reset by peer
Apr 20 17:00:07 kind-1.17.0-control-plane kubelet[8244]: I0420 17:00:07.079624    8244 prober.go:111] Liveness probe for "virt-launcher-testvmib2nfdbzw67zkn6svftrlh2tg7zrdcgr2wgw6lwz86x_kubevirt-test-default(c64c9157-c7a1-43ff-ba10-64468e7b3415):compute" errored: read tcp [fd00:10:244::1]:59148->[fd00:10:244::28]:1500: read: connection reset by peer
Apr 20 17:00:12 kind-1.17.0-control-plane kubelet[8244]: I0420 17:00:12.080025    8244 prober.go:111] Liveness probe for "virt-launcher-testvmib2nfdbzw67zkn6svftrlh2tg7zrdcgr2wgw6lwz86x_kubevirt-test-default(c64c9157-c7a1-43ff-ba10-64468e7b3415):compute" errored: read tcp [fd00:10:244::1]:59184->[fd00:10:244::28]:1500: read: connection reset by peer
Apr 20 17:00:17 kind-1.17.0-control-plane kubelet[8244]: I0420 17:00:17.079803    8244 prober.go:111] Liveness probe for "virt-launcher-testvmib2nfdbzw67zkn6svftrlh2tg7zrdcgr2wgw6lwz86x_kubevirt-test-default(c64c9157-c7a1-43ff-ba10-64468e7b3415):compute" errored: read tcp [fd00:10:244::1]:59264->[fd00:10:244::28]:1500: read: connection reset by peer
Apr 20 17:00:22 kind-1.17.0-control-plane kubelet[8244]: I0420 17:00:22.079562    8244 prober.go:111] Liveness probe for "virt-launcher-testvmib2nfdbzw67zkn6svftrlh2tg7zrdcgr2wgw6lwz86x_kubevirt-test-default(c64c9157-c7a1-43ff-ba10-64468e7b3415):compute" errored: read tcp [fd00:10:244::1]:59300->[fd00:10:244::28]:1500: read: connection reset by peer
Apr 20 17:00:27 kind-1.17.0-control-plane kubelet[8244]: I0420 17:00:27.079915    8244 prober.go:111] Liveness probe for "virt-launcher-testvmib2nfdbzw67zkn6svftrlh2tg7zrdcgr2wgw6lwz86x_kubevirt-test-default(c64c9157-c7a1-43ff-ba10-64468e7b3415):compute" errored: read tcp [fd00:10:244::1]:59358->[fd00:10:244::28]:1500: read: connection reset by peer
Apr 20 17:00:32 kind-1.17.0-control-plane kubelet[8244]: I0420 17:00:32.079316    8244 prober.go:111] Liveness probe for "virt-launcher-testvmib2nfdbzw67zkn6svftrlh2tg7zrdcgr2wgw6lwz86x_kubevirt-test-default(c64c9157-c7a1-43ff-ba10-64468e7b3415):compute" errored: read tcp [fd00:10:244::1]:59394->[fd00:10:244::28]:1500: read: connection reset by peer
```
One thing that can be seen is that the readiness check really does keep the pod out of the ready state, and that's what we're checking in the [test code](https://github.com/kubevirt/kubevirt/blob/51ce9d8e6def33c0c260f0cdb994f08ae5b9938e/tests/probes_test.go#L73).

As for liveness, I fail to understand how it tolerates the failed messages. For that, I'll need to dig k8s code.